packages/flutter_tools/lib/src/isolated/native_assets/native_assets.dart

```diff
@@ -409,9 +343,9 @@ Future<void> ensureNoNativeAssetsOrOsIsSupported(
 
 /// This should be the same for different archs, debug/release, etc.
 /// It should work for all macOS.
-Uri nativeAssetsBuildUri(Uri projectUri, OS os) {
+Uri nativeAssetsBuildUri(Uri projectUri, String osName) {
   final String buildDir = getBuildDirectory();
-  return projectUri.resolve('$buildDir/native_assets/$os/');
+  return projectUri.resolve('$buildDir/native_assets/$osName/');
```

Comment:
Nit: use `fileSystem.path.join(...)` rather than building paths manually.

Reply:
(I tried and reverted this - it doesn't match the use of URIs everywhere else, leading to windows path problems)